File: slack-poster/src/test/java/dev/number6/slackposter/dagger/FakeHttpModule.java
package dev.number6.slackposter.dagger;

import dev.number6.slack.port.HttpPort;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class FakeHttpModule {

    @Provides
    @Singleton
    public HttpPort providesHttp() {
        return new RecordingHttpAdaptor();
    }
}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/dagger/FakeHttpModule.kt
package dev.number6.slackposter.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slack.port.HttpPort
import javax.inject.Singleton

@Module
class FakeHttpModule {
    @Provides
    @Singleton
    fun providesHttp(): HttpPort {
        return RecordingHttpAdaptor()
    }
}